server/src/main/java/org/mixit/cesar/site/web/api/ArticleController.java

```diff
@@ -23,14 +29,38 @@
     @Autowired
     ArticleRepository articleRepository;
 
+    @Autowired
+    private HttpServletRequest httpServletRequest;
+
+    @Autowired
+    private AccountRepository accountRepository;
+
     @RequestMapping("/{id}")
     @ApiOperation(value = "Finds one article", httpMethod = "GET")
     @JsonView(FlatView.class)
     public ResponseEntity<Article> getArticle(@PathVariable("id") Long id) {
+        Article result;
+        if (isCurrentUserAdmin()) {
+            result = articleRepository.findArticleById(id);
+        } else {
+            result = articleRepository.findPublishedArticleById(id);
+        }
+
         return ResponseEntity
                 .ok()
-                //.cacheControl(CacheControl.maxAge(4, TimeUnit.DAYS))
-                .body(articleRepository.findPublishedArticleById(id));
+                .cacheControl(CacheControl.noCache())
+                .body(result);
+    }
+
+    private boolean isCurrentUserAdmin() {
+        String token = httpServletRequest.getHeader(AuthenticationFilter.TOKEN_REQUEST_HEADER_PARAM);
```

Comment:
I changed the filter to authenticate users. If you make a rebase you can use the currentUser stored in applicationContext